llama_hub/llama_packs/resume_screener/base.py

```diff
@@ -0,0 +1,68 @@
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+from llama_index.llama_pack.base import BaseLlamaPack
+from llama_index.readers import PDFReader
+from llama_index.llms.base import LLM
+from llama_index.llms import OpenAI
+from llama_index import ServiceContext
+from llama_index.schema import NodeWithScore
+from llama_index.response_synthesizers import TreeSummarize
+
+from pydantic import BaseModel, Field
+
+QUERY_TEMPLATE = """
+You are an expert resume reviewer. 
+You job is to decide if the candidate pass the resume screen given the job description and a list of criteria:
+
+### Job Description
+{job_description}
+
+### Screening Criteria
+{criteria_str}
+"""
+
+class CriteriaDecision(BaseModel):
+    """The decision made based on a single criteria"""
+    decision: bool = Field(description="The decision made based on the criteria")
+    reasoning: str = Field(description="The reasoning behind the decision")
+
+
+class ResumeScreenerDecision(BaseModel):
+    """The decision made by the resume screener"""
+    criteria_decisions: List[CriteriaDecision] = Field( description="The decisions made based on the criteria")
+    overall_reasoning: str = Field(description="The reasoning behind the overall decision")
+    overall_decision: bool = Field(description="The overall decision made based on the criteria")
+
+
+def _format_criteria_str(criteria: List[str]) -> str:
+    criteria_str = ""
+    for criterion in criteria:
+        criteria_str += f"- {criterion}\n"
+    return criteria_str
+
+class ResumeScreenerPack(BaseLlamaPack):
+    def __init__(self, job_description: str, criteria: List[str], llm: Optional[LLM] = None) -> None:
+        self.reader = PDFReader()
+        llm = llm or OpenAI(model='gpt-4')
+        service_context = ServiceContext.from_defaults(llm=llm)
+        self.synthesizer = TreeSummarize(output_cls=ResumeScreenerDecision, service_context=service_context)
```

Comment:
I wonder if the `Refine` synthesizer is a better default? Not really sure tbh

Reply:
Made this configurable, but still defaults to tree summarize.